derp/derp_server_linux.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package derp
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+func (c *sclient) statsLoop(ctx context.Context) error {
+	// If we can't get a TCP socket, then we can't send stats.
+	tcpConn, ok := c.nc.(*net.TCPConn)
+	if !ok {
+		// TODO(andrew): metric?
+		return nil
```

Comment:
Log the error and/or add metric to the label map?

Reply:
👍